### ISSUE TITLE
Preserve reexports in a tree-shakeable form

### ISSUE THREAD

**Issue by Andarist** [rollup/rollup#5161]
### Feature Use Case

I'd like to keep my libraries as tree-shakeable as possible

### Feature Proposal

As we can see here:
[Rollup playground](https://rollupjs.org/repl/?version=3.29.4&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBhcyUyMG1hdGhzJTIwJTIwZnJvbSUyMCcuJTJGbWF0aHMuanMnJTNCJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTJDJTIybmFtZSUyMiUzQSUyMm1haW4uanMlMjIlN0QlMkMlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwY29uc3QlMjBzcXVhcmUlMjAlM0QlMjB4JTIwJTNEJTNFJTIweCUyMColMjB4JTNCJTVDbmV4cG9ydCUyMGNvbnN0JTIwY3ViZSUyMCUzRCUyMHglMjAlM0QlM0UlMjB4JTIwKiUyMHglMjAqJTIweCUzQiU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlMkMlMjJuYW1lJTIyJTNBJTIybWF0aHMuanMlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyb3V0cHV0JTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlN0QlMkMlMjJ0cmVlc2hha2UlMjIlM0F0cnVlJTdEJTdE)

we end up with a frozen object being exported because everything gets flat-bundled and because we used:
```ts
export * as maths  from './maths.js';
```

It would be great if a chunk could be (optionally?) created for all of such reexports to keep the code in a format that doesn't deoptimize most of the tree-shakers out there.

As an added bonus the same could be applied to:
```ts
import * as maths  from './maths.js';
export { maths };
```

**Comment by 0x009922** [comment]
Might be a duplicate of

- https://github.com/rollup/rollup/issues/4907

**Comment by lukastaegert** [comment]
No, that one is slightly different and could be considered a bug (but it is hard to fix for technical reasons). The current proposal is to create additional chunks to replace "fake" namespace objects with actual namespaces.

**Comment by samualtnorman** [comment]
are there any known workarounds for this?

**Comment by teidesu** [comment]
the only workaround i found for this is using `output.preserveModules`, but this disables any bundling at all which is definitely not ideal :c

**Comment by samualtnorman** [comment]
> the only workaround i found for this is using `output.preserveModules: true`, but this disables any bundling at all which is definitely not ideal :c

yeah that's what I've been using but it comes with quite the drawback
